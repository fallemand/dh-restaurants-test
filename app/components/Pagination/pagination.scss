@import '../../styles/variables';

.pagination {
  display: inline-block;

  .pagination__number {
    color: $font-color-primary;
    font-weight: $font-light;
    float: left;
    font-size: 13px;
    padding: 8px 16px;
    text-decoration: none;
    transition: background-color .3s;
    border: 1px solid $background-color-tertiary;
    margin: 0 4px;

    &.active {
      background-color: $background-color-primary;
      color: white;
      border: 1px solid $background-color-primary;
    }

    &:hover:not(.active) {
      background-color: $background-color-tertiary;
    }
  }
}
