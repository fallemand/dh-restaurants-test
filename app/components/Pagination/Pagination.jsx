import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import './pagination.scss';

const Pagination = ({ className, total, show }) => {
  const pages = Math.ceil(total / show);
  return (
    <div className={classnames(className, 'pagination')}>
      <a className="pagination__number" href="#/">&laquo;</a>
      {Array.from(Array(pages), (x, i) => i).map(
        pageNumber => <a className="pagination__number" href="/#">{pageNumber}</a>,
      )}
      <a className="pagination__number" href="#/">&raquo;</a>
    </div>
  );
}

Pagination.propTypes = {
  className: PropTypes.string,
  total: PropTypes.number.isRequired,
  show: PropTypes.number.isRequired,
};

Pagination.defaultProps = {
  className: null,
};

module.exports = Pagination;
