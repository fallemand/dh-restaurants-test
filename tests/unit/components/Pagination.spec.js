import React from 'react';
import Pagination from '../../../app/components/Pagination';

describe('Pagination', () => {
  const props = {
    total: 50,
    show: 5,
    className: '__CLASSNAME__',
  };

  it('render component', () => {
    const component = shallow(<Pagination {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('count amount of pages generated', () => {
    const component = shallow(<Pagination {...props} />);
    const buttons = component.find('.pagination__number');
    expect(buttons).toHaveLength((props.total / props.show) + 2);
  });
});
