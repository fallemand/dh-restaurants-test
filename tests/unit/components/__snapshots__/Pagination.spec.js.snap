// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render component 1`] = `
"<div className='__CLASSNAME__ pagination'>
  <a className='pagination__number' href='#/'>
    «
  </a>
  <a className='pagination__number' href='/#'>
    0
  </a>
  <a className='pagination__number' href='/#'>
    1
  </a>
  <a className='pagination__number' href='/#'>
    2
  </a>
  <a className='pagination__number' href='/#'>
    3
  </a>
  <a className='pagination__number' href='/#'>
    4
  </a>
  <a className='pagination__number' href='/#'>
    5
  </a>
  <a className='pagination__number' href='/#'>
    6
  </a>
  <a className='pagination__number' href='/#'>
    7
  </a>
  <a className='pagination__number' href='/#'>
    8
  </a>
  <a className='pagination__number' href='/#'>
    9
  </a>
  <a className='pagination__number' href='#/'>
    »
  </a>
</div>"
`;
